src/RequestInformation.cs

```diff
@@ -49,20 +49,21 @@ public Uri URI
                     if(UrlTemplate?.IndexOf("{+baseurl}", StringComparison.OrdinalIgnoreCase) >= 0 && !PathParameters.ContainsKey("baseurl"))
                         throw new InvalidOperationException($"{nameof(PathParameters)} must contain a value for \"baseurl\" for the url to be built.");
 
-                    var parsedUrlTemplate = new UriTemplate(UrlTemplate, caseInsensitiveParameterNames: true);
```

Comment:
This is effectively the same problem as Go and we should instead tweak the generation to add the missing attribute.
https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/b0cba5897832c86c37e758184c18ef055c82c1f8/src/Microsoft.Graph/Generated/Groups/Item/CalendarView/Item/Instances/Delta/DeltaRequestBuilder.cs#L144

Reply:
I'm not sure I understand what the bug is, but yes, in the go implementation, there is a lowering of all the keys which is suspicious ...

Reply:
It's the same issue as Go, we're uppercasing the properties in dotnet and using reflection to get the name (with uppercase) which might not match the name in the template. Dotnet didn't do a normalization like go because the previous lib has a setting to perform a case insensitive expansion.

Reply:
Got it, understood and I can replicate the issue, working on it if no-one else is picking it up.

Reply:
please go ahead!